### ISSUE TITLE
FULL UPDATE AND NO WORKY WORKY

### ISSUE THREAD

**Issue by CHILLYSMOKES** [KevinOConnor/klipper#3081]
Dual MCU: 1- RE-ARM LPC1768 (runs everything but LCD), 2- Arduino

After a year I decided to update. Updated Rpi, Klipper, MCU Firmware... no joy.  Hoping someone here can help. If I have missed something in my research I apologize but I spent all day on this.  Log is attached.
[klippy.log](https://github.com/KevinOConnor/klipper/files/4908018/klippy.log)

-made the some changes to printer.cfg, renaming stuff that was changed and my usb serial ID changed so I changed that to match the output of "ls /dev/serial/by-id/*".  Changed it back.  Changed it many times in "make menuconfig" and matched it in printer.cfg but nothing worked.
-restarted firmware, host, rpi... nothing.

This is the error in OctoPrint I get:
![problem1](https://user-images.githubusercontent.com/36556452/87239272-2af51d00-c3c2-11ea-8f12-e45a130ebf03.jpg)

Thank you very much for looking at this.  Thank you thank you.

**Comment by CHILLYSMOKES** [comment]
After more research I believe it is a communication deal methinks.
Reran update and noticed some things.
-pip wasn't getting updated because directory in install-octopi.sh pointing to it was incorrect described here #3011 
     -went into install-octopi.sh and manually changed PYTHONDIR
     -reran update and it gets further but there is still an error about updating some portions of python
-#261 reset the virtual environment and then rerun update
     -update gets much further with only a note at the end that something went wrong or wasn't found.
-Now I think I'm making this code look all over the place, any future updates will get all messed up.
-uninstalled Klipper service and all dir and klipper-env.
-reinstalled Klipper
-make menuconfig with default settings, changing only my MCU
-back to the same issue in pic above....

**Comment by CHILLYSMOKES** [comment]
Found it.

I removed the second MCU lines in printer.cfg thinking its gotta be messing up something or I missed something with how adding another MCU into the equation got changed somewhere.  Also, unplugged the USB from the pi.  Klipper is working.  I'll troubleshoot the additional MCU and LCD later.  I'll close this.  Hopefully, this helps someone else without having to evoke an ISSUE.  For that I include the following text for search purposes.  

"Error during identify: cannot convert dictionary update sequence element #0 to a sequence This type of error is frequently caused by running an older version of the firmware on the micro-controller (fix by recompiling and flashing the firmware). Once the underlying issue is corrected, use the "RESTART" command to reload the config and restart the host software. Protocol error connecting to printer Klipper state: Not ready"

REMOVE ANY ADDITIONAL MCU FROM PRINTER.CFG AND UNPLUG FROM THE PI.  ISSUE COMMAND SUDO SERVICE KLIPPER RESTART.
